tests/migrations/test_writer.py

```diff
@@ -12,12 +12,11 @@
 import custom_migration_operations.operations
 
 from django import get_version
-from django.conf import settings
+from django.conf import SettingsReference, settings
 from django.core.validators import EmailValidator, RegexValidator
 from django.db import migrations, models
-from django.db.migrations.writer import (
-    MigrationWriter, OperationWriter, SettingsReference,
-)
+from django.db.migrations.serializer import BaseSerializer
+from django.db.migrations.writer import MigrationWriter, OperationWriter
 from django.test import SimpleTestCase
```

Comment:
I fixed these import but ci tool gave an error about isort. Is this about me or a cache problem.

Reply:
I think that there have just been some issues with pulling out changes into separate commits. I have done some tidying up for you in [a branch](https://github.com/pope1ni/django/commits/ticket-29738) to separate the three commits properly and fixed a bunch of issues with the imports and tests too.

I think that this now needs some documentation in the second commit if we are going to make this public API. (The third commit doesn't need any documentation as it counts as a bug fix.)

I feel that this is close to completion and it would be nice to get it in ahead of the feature freeze for 2.2 next week. Are you able to progress this?

Reply:
Of course, I look forward to it. 
Is there any way to merge your commits with a git command onto my commits? If it is not possible, I will use the copy-paste method.

Reply:
Should we add just one sentence into 2.2.txt?

--- a/docs/releases/2.2.txt
+++ b/docs/releases/2.2.txt
@@ -197,6 +197,8 @@ Migrations

 * ``NoneType`` can now be serialized in migrations.

+*  Allowed registering serializers with ``MigrationWriter``.
+
 Models
 ~~~~~~

Reply:
A release note, yes, but that should link to an explanation in topics/migrations about how to use `MigrationWriter.register_serializer()`.